kratos/tests/test_model_part.py

```diff
@@ -64,7 +64,36 @@ def test_model_part_sub_model_parts(self):
 
         self.assertFalse(model_part.HasSubModelPart("Inlets"))
         self.assertEqual(model_part.NumberOfSubModelParts(), 0)
-        #print (model_part)
+
+    def test_clear_model_part(self):
+        current_model = Model()
+
+        model_part= current_model.CreateModelPart("Main")
+        model_part.CreateSubModelPart("Inlets")
+        model_part.CreateSubModelPart("Temp")
+        model_part.CreateSubModelPart("Outlet")
+
+        self.assertEqual(model_part.NumberOfSubModelParts(), 3)
+        sub_model_part_1 = model_part.GetSubModelPart("Inlets")
+        sub_model_part_1.CreateSubModelPart("Inlet1")
+        sub_model_part_1.CreateSubModelPart("Inlet2")
+
+        model_part.GetSubModelPart("Outlet").CreateSubModelPart("sub_outlet1")
+        self.assertEqual(model_part.GetSubModelPart("Outlet").NumberOfSubModelParts(), 1)
+        model_part.GetSubModelPart("Outlet").Clear()
+        self.assertEqual(model_part.GetSubModelPart("Outlet").NumberOfSubModelParts(), 0)
+
+        model_part.Set(SLAVE)
+        self.assertTrue(model_part.Is(SLAVE))
+ 
+        #here the 
+        model_part.Clear()
+ 
+        self.assertEqual(model_part.NumberOfSubModelParts(),0)
+        self.assertEqual(len(model_part.Nodes),0)

```

Comment:
Are nodes created at any point in the test? What should we expect to happen in that case? If I clear a submodelpart, will it delete its nodes/elements from its parent(s)?

Reply:
Nodes are stored in meshes, and i delete the meshes, so what i am deleting are the smart_ptrs to the nodes. they are definitely not deleted from the father...

having said this, it is true that i am not creating nodes here. I should modify the test to include nodes creation

Reply:
Thanks, if you could do that it would be very useful as a minimal example of usage for the new features.

Reply:
@jcotela  dones. Also i am now calling clear on the Communicator which i think is to be expected